### PR TITLE
G.h to CDC, WHO comparisons

### PR DESCRIPTION
**Pull request by jim-sheldon**
Added script to pull data, compare to G.h, update sheet
Renamed folders, added directory for templating utility scripts

### CODE REVIEW

File: scripts/cdc_who_gh_comparison/run.py [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+
+def change_gh_data(changes: dict[str, int], dry_run: bool):
+	logging.info("Changing G.h USA data")
+	client = pygsheets.authorize(service_account_env_var="GOOGLE_CREDENTIALS")
+	client.set_batch_mode(True)
+	spreadsheet = client.open_by_key(DOCUMENT_ID)
+
+	sheet = spreadsheet.worksheet("title", LINE_LIST_SHEET)
+
+	# Finding column indices based on values breaks when using enums...
```
Comment (abhidg) [inline scripts/cdc_who_gh_comparison/run.py:199]: This is likely to break -- is there a workaround to use column names?

File: scripts/cdc_who_gh_comparison/run.py [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+		"Status": "confirmed",
+		"Date_confirmation": today,
+		"Date_entry": today,
+		"Date_last_modified": today,  # FIXME: seeing "44782" in sheet
+		"Source": CDC_SOURCE
+	})
+	return output.getvalue().removesuffix("\r\n")
+
+
+def find_extra_rows(records, state, count):
```
Comment (abhidg) [inline scripts/cdc_who_gh_comparison/run.py:248]: types?

**Review by abhidg — COMMENTED** [review]
Looks OK, I'd add some tests to check that updating the spreadsheet works correctly

**Comment by abhidg** [comment]
@tvarrelman This script currently assumes that US cases are fungible for a particular state, is that likely to continue? I am thinking of a scenario where there might be info added to a particular case (symptoms); and CDC reports two less cases than us, and we set these two to omit_error, one of which is actually confirmed and has more info such as symptoms. OTOH, if all US cases now, only have location, date and status, then this is OK

File: scripts/cdc_who_gh_comparison/run.py (hunk 1 above)
Comment (jim-sheldon) [reply scripts/cdc_who_gh_comparison/run.py:199]: No, unfortunately this is the workaround for the `find` function breaking

**Comment by jim-sheldon** [comment]
> Looks OK, I'd add some tests to check that updating the spreadsheet works correctly

Agreed, I am working on testing it now.

File: scripts/cdc_who_gh_comparison/run.py (hunk 2 above)
Comment (jim-sheldon) [reply scripts/cdc_who_gh_comparison/run.py]: Oops, forgot a few. Added.

**Comment by jim-sheldon** [comment]
> @tvarrelman This script currently assumes that US cases are fungible for a particular state, is that likely to continue? I am thinking of a scenario where there might be info added to a particular case (symptoms); and CDC reports two less cases than us, and we set these two to omit_error, one of which is actually confirmed and has more info such as symptoms. OTOH, if all US cases now, only have location, date and status, then this is OK

This is a great point. Should I add any checks for other columns like symptoms, and only change cases to "omit_error" when those columns are empty?

**Comment by jim-sheldon** [comment]
I refactored scripts into their own directory with shared building and testing strategies and files. It should now be easy to add new ones in a new subdirectory. In the future we could add a tool like [cookiecutter](https://cookiecutter.readthedocs.io) to make it even easier.

**Review by abhidg — APPROVED** [review]

**Pull request merged**